### PR TITLE
[fix] Resolve scikit-learn `check_array` deprecation warning 

### LINKED ISSUE #117: Future warning from scikit-learn
Use of `MoleculeTranformer` produces future warnings from scikit-learn. The `transform` method of `MoleculeTransformer` ends with a call to `molfeat.utils.datatype.as_numpy_array_if_possible`, which relies on `sklearn.utils.check_array`. Currently, the keyword argument `force_all_finite` is used here, but this is being deprecated in favor of `ensure_all_finite`.

### PR DESCRIPTION
**Pull request by smcolby**
Resolves #117 

## Changelogs

- Resolve deprecation/future warning by changing `force_all_finite` to `ensure_all_finite` in call to `sklearn.utils.check_array`

---

_Checklist:_

- [X] _Was this PR discussed in an issue? It is recommended to first discuss a new feature into a GitHub issue before opening a PR._
- [X] _Add tests to cover the fixed bug(s) or the new introduced feature(s) (if appropriate)._
- [X] _Update the API documentation if a new function is added, or an existing one is deleted. Eventually consider making a new tutorial for new features._
- [X] _Write concise and explanatory changelogs below._
- [X] _If possible, assign one of the following labels to the PR: `feature`, `fix` or `test` (or ask a maintainer to do it for you)._

---

### CODE REVIEW

**Comment by maclandrol** [comment]
Thanks for the scikit-learn compatibility fix. The original commit is now included in dev for the next major release. GitHub cannot retarget the PR because the branch already contains it, so I am closing it as integrated.
